### ISSUE TITLE
[Truffle] Super in an aliased method can confuse the ancestor chain

### ISSUE THREAD

**Issue by chrisseaton** [jruby/jruby#3604]
```ruby
class Foo
  def foo
    puts 'Foo#foo'
  end
end

class Bar < Foo
  def foo
    puts 'Bar#foo'
    super
  end
end

class Dod < Bar
  alias_method :bar, :foo
end

Dod.new.bar
```

Expected

```
Bar#foo
Foo#foo
```

Actual

```
Bar#foo
Bar#foo
Foo#foo
```

**Comment by eregon** [comment]
The lookup is wrong because the definition module of :bar is Dod. We need to find the proper rules for super lookup, maybe we need to keep track of the actual module where the method was defined and not the module to which the method was attached.

**Comment by eregon** [comment]
I took a quick look and it seems that keeping the definition module to be where the method was lexically defined works.

**Comment by eregon** [comment]
Fixed by dafc4601d30ee144e215144eb9f0ac620e1dec07.
